Rendera/include/scene.hpp

```diff
@@ -116,43 +126,92 @@ class Scene{
                     hitinfo[y*img_width + x].ishitting = true;
                     hitinfo[y*img_width + x].dist = i_.dist_1();
                     hitinfo[y*img_width + x].intersection_pt = hitinfo[y*img_width + x].ray_.fetch(hitinfo[y*img_width + x].dist); 
-                    hitinfo[y*img_width + x].normal = hitable_sphere[i].normal(hitinfo[y*img_width + x].intersection_pt);
-                    hitinfo[y*img_width + x].shade = lighting(hitable_sphere[i].mat(), hitinfo[y*img_width + x].ray_, hitinfo[y*img_width + x].normal, hitinfo[y*img_width + x].intersection_pt, light_src, light_col);
+                    hitinfo[y*img_width + x].normal = (*sp).normal(hitinfo[y*img_width + x].intersection_pt);
+                    hitinfo[y*img_width + x].shade = lighting((*sp).mat(), hitinfo[y*img_width + x].ray_, hitinfo[y*img_width + x].normal, hitinfo[y*img_width + x].intersection_pt, light_src, light_col);
+                    hitinfo[y*img_width + x].shade += lighting((*sp).mat(), hitinfo[y*img_width + x].ray_, hitinfo[y*img_width + x].normal, hitinfo[y*img_width + x].intersection_pt, light_src2, light_col);
+                    hitinfo[y*img_width + x].rgb = hitinfo[y*img_width + x].shade.get_int();
+                    hitinfo[y*img_width + x].ref_ray = reflect(hitinfo[y*img_width + x].ray_, hitinfo[y*img_width + x].normal);
+
+                }
+            }
+
+            for(cb = hitable_cube.begin(); cb != hitable_cube.end(); ++cb){
+                bool hits = (*cb).hit(hitinfo[y*img_width + x].ray_);
+                if(hitinfo[y*img_width + x].ishitting && hits){
+                    float d = (*cb).get_int_pt().norm();
+                    if (hitinfo[y*img_width + x].dist>d){
+                        hitinfo[y*img_width + x].ishitting = true;
```

Comment:
Same thing here. If you are writing massive walls of text like this, you are doing something wrong.